crates/e2e/src/setup/solver/solution.rs

```diff
@@ -45,47 +45,58 @@ impl JitOrder {
         signing_scheme: EcdsaSigningScheme,
         domain: &DomainSeparator,
         key: SecretKeyRef,
-    ) -> solvers_dto::solution::JitOrder {
+    ) -> (solvers_dto::solution::JitOrder, OrderUid) {
         let data = self.data();
-        let signature = match model::signature::EcdsaSignature::sign(
+        let signature = model::signature::EcdsaSignature::sign(
             signing_scheme,
             domain,
             &data.hash_struct(),
             key,
         )
-        .to_signature(signing_scheme)
-        {
+        .to_signature(signing_scheme);
+        let order_uid = data.uid(
+            domain,
+            &signature
+                .recover_owner(&signature.to_bytes(), domain, &data.hash_struct())
+                .unwrap(),
+        );
+        let signature = match signature {
             model::signature::Signature::Eip712(signature) => signature.to_bytes().to_vec(),
             model::signature::Signature::EthSign(signature) => signature.to_bytes().to_vec(),
             model::signature::Signature::Eip1271(signature) => signature,
             model::signature::Signature::PreSign => panic!("Not supported PreSigned JIT orders"),
         };
-        solvers_dto::solution::JitOrder {
-            sell_token: data.sell_token,
-            buy_token: data.buy_token,
-            receiver: data.receiver.unwrap_or_default(),
-            sell_amount: data.sell_amount,
-            buy_amount: data.buy_amount,
-            valid_to: data.valid_to,
-            app_data: data.app_data.0,
-            kind: match data.kind {
-                OrderKind::Buy => Kind::Buy,
-                OrderKind::Sell => Kind::Sell,
-            },
-            sell_token_balance: match data.sell_token_balance {
-                SellTokenSource::Erc20 => solvers_dto::solution::SellTokenBalance::Erc20,
-                SellTokenSource::External => solvers_dto::solution::SellTokenBalance::External,
-                SellTokenSource::Internal => solvers_dto::solution::SellTokenBalance::Internal,
-            },
-            buy_token_balance: match data.buy_token_balance {
-                BuyTokenDestination::Erc20 => solvers_dto::solution::BuyTokenBalance::Erc20,
-                BuyTokenDestination::Internal => solvers_dto::solution::BuyTokenBalance::Internal,
+        (
```

Comment:
Why do we have a 32 lines tuple instead of a named variable for the `jit_order` and a single line tuple. This is unnecessarily hard to read and increases the diff for no good reason.